splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/BroadcastJoinOperation.java

```diff
@@ -270,7 +294,10 @@ else if (notExistsRightSide)
             }
         }
 
-        result = result.map(new CountProducedFunction(operationContext), /*isLast=*/true);
+        // Counting rows in this manner disables native spark execution,
+        // so only do it for non-native spark.
+        if (!usesNativeSparkDataSet)
+            result = result.map(new CountProducedFunction(operationContext), /*isLast=*/true);
```

Comment:
You can add 
`
    @Override
    public boolean hasNativeSparkImplementation() {
        return true;
    }`

To CountProducedFunction so that it gets ignored in a NativeSparkDataset. 

In the JoinRestrictionPredicateFunction you could return true if restriction is != null for instance.

Reply:
OK, I'll add hasNativeSparkImplementation to CountProducedFunction.  For JoinRestrictionPredicateFunction, returning true without providing an implementation of a native spark join filter would cause queries to return incorrect results.  I have opened [SPLICE-2362](https://splice.atlassian.net/browse/SPLICE-2362) to handle the native spark join filter.